src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -522,8 +522,18 @@ internal static string RemoveGuidFromMessage(string message, out bool matchPatte
         internal static string RemoveIdentifierInfoFromMessage(string message, out bool matchPattern)
         {
             matchPattern = false;
-            if (string.IsNullOrEmpty(message))
+            if (message is null ||
+                message.Length < 40 ||
+                message[37] != '[' ||
+                message[36] != ':' ||
+                message[23] != '-' ||
+                message[18] != '-' ||
+                message[13] != '-' ||
+                message[8] != '-')
```

Comment:
```suggestion
                message[8] != '-' ||
                message[13] != '-' ||
                message[18] != '-' ||
                message[23] != '-' ||
                message[36] != ':' ||
                message[37] != '[')
```

Since we check the length, it is not necessary to check in reverse order to remove a bounds check.

Reply:
My intention is another - check the least likely characters first.

Indexes were fixed. Thanks!